app/src/main/java/com/example/bookwormadventuresdeluxe2/FirebaseUserGetSet.java

```diff
@@ -0,0 +1,186 @@
+package com.example.bookwormadventuresdeluxe2;
+
+import android.util.Log;
+import android.widget.EditText;
+
+import androidx.annotation.NonNull;
+
+import com.example.bookwormadventuresdeluxe2.Utilities.EditTextValidator;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.FirebaseUser;
+
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static android.content.ContentValues.TAG;
+
+/**
+ * Firebase writer and reader for UserProfileObject
+ */
+public class FirebaseUserGetSet
+{
+    private static FirebaseFirestore firebase = FirebaseFirestore.getInstance();
+    private static CollectionReference usersRef = firebase.collection("Users");
+
+    /**
+     * Performs query to extract UserProfileObject from database
+     *
+     * @param username username of object to be called
+     * @param myCallback interface for returning object after query success
+     */
+    public static void getUser(String username, UserCallback myCallback)
+    {
+        usersRef.whereEqualTo("username", username).get()
+                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>()
+                {
+                    @Override
+                    public void onComplete(@NonNull Task<QuerySnapshot> task)
+                    {
+                        if (task.isSuccessful())
+                        {
+                            for (QueryDocumentSnapshot document : task.getResult())
+                            {
+                                /* Extracting userObject from document */
+                                UserProfileObject userObject = new UserProfileObject(
+                                                    document.getData().get("username").toString(),
+                                                    document.getData().get("email").toString(),
+                                                    document.getData().get("userId").toString(),
+                                                    document.getId()
+                                                    );
+
+                                if (document.getData().get("phoneNumber").toString() != null)
```

Comment:
Phone numbers should be always mandatory from a question asked to the prof, why is this being handled any different than the other fields?

Reply:
ok